### DISCUSSION TITLE
setRequestHeader fail on XMLHttpRequestOverride

### DISCUSSION THREAD

**Discussion by eilon-shai** [mswjs/msw#2222] [Q&A] [answered]
Hi, 

I am using msw in our tests and I cant understand how the functionality of 
setRequestHeader can work on XMLHttpRequestOverride.
in my code I am creating a new XMLHttpRequest() and that become XMLHttpRequestOverride object in the tests instead of XMLHttpRequest.

then in the setRequestHeader code of XMLHttpRequest.js I am failing on the:

`exports.is = value => {
  return utils.isObject(value) && utils.hasOwn(value, implSymbol) && value[implSymbol] instanceof Impl.implementation;
};`

There is a constraint to use the original implementation.

any idea how can I overcome this?

Thanks,
Eilon

**Comment by kettanaito** [comment, marked as answer]
Hi, @eilon-shai. Thanks for reporting this. 

Could you please open a issue about this and provide a minimal reproduction repo for me to look at? I'd be able to look into the issue then. Thanks.
